pkg/ddl/backfilling.go

```diff
@@ -388,6 +388,11 @@ func splitTableRanges(t table.PhysicalTable, store kv.Storage, startKey, endKey
 		zap.Int64("physicalTableID", t.GetPhysicalID()),
 		zap.String("start key", hex.EncodeToString(startKey)),
 		zap.String("end key", hex.EncodeToString(endKey)))
+	if len(startKey) == 0 && len(endKey) == 0 {
+		logutil.BgLogger().Info("split table range from PD, get noop table range", zap.String("category", "ddl"), zap.Int64("physicalTableID", t.GetPhysicalID()))
+		return []kv.KeyRange{}, nil
+	}
+
```

Comment:
Why not check this in origin place? Thus we can skip `setupWorkers()` and `consumer.run()`

Reply:
Because there will be return an error as described in https://github.com/pingcap/tidb/issues/49682